src/client.ts

```diff
@@ -4627,22 +4627,24 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             threadId = null;
         }
         const reason = opts?.reason;
+        let withRelTypes = opts?.with_rel_types ?? opts?.with_relations;
 
-        if (
-            opts?.with_relations &&
-            this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported
-        ) {
+        if (typeof withRelTypes === "string") {
+            withRelTypes = [withRelTypes];
+        }
+
+        if (withRelTypes && this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported) {
```

Comment:
implicit cast to bool is forbidden by our code style.

Likewise below

Reply:
Explicit cast to bool is forbidden by ESLint :clown_face: 

```
4633:13  error  Redundant double negation  no-extra-boolean-cast
```

Reply:
Well, that's new. Surely we use double-negation everywhere?

I don't consider this resolved: we can't just ignore the style guide. Either we agree as a team to change it, or we respect it.

Anyway, in this case an explicit test for `undefined` is more appropriate.